packages/dates/src/utilities/formatDate.ts

```diff
@@ -1,16 +1,38 @@
+import memoize from 'lodash/memoize';
+
 export function formatDate(
   date: Date,
-  locales: string,
+  locales: string | string[],
   options: Intl.DateTimeFormatOptions = {},
-): string {
+) {
   // Etc/GMT+12 is not supported in most browsers and there is no equivalent fallback
   if (options.timeZone != null && options.timeZone === 'Etc/GMT+12') {
     const adjustedDate = new Date(date.valueOf() - 12 * 60 * 60 * 1000);
 
-    return Intl.DateTimeFormat(locales, {...options, timeZone: 'UTC'}).format(
-      adjustedDate,
-    );
+    return memoizedGetDateTimeFormat(locales, {
+      ...options,
+      timeZone: 'UTC',
+    }).format(adjustedDate);
   }
 
-  return Intl.DateTimeFormat(locales, options).format(date);
+  return memoizedGetDateTimeFormat(locales, options).format(date);
+}
+
+function getDateTimeFormat(
+  locales?: string | string[],
+  options?: Intl.DateTimeFormatOptions,
+) {
+  return Intl.DateTimeFormat(locales, options);
 }
+
+function dateTimeFormatCacheKey(
+  locales: string | string[],
+  options?: Intl.DateTimeFormatOptions,
+) {
+  return `${locales}-${JSON.stringify(options)}`;
+}
+
+const memoizedGetDateTimeFormat = memoize(
```

Comment:
Should this `const` be defined before it's used?

Reply:
It works but harder for reading.
I will update.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#Function_scope
> a function can access all variables and functions defined inside the scope in which it is defined. In other words, a function defined in the global scope can access all variables defined in the global scope.